### DISCUSSION TITLE
Save deconvoluted spectrum during  a batch run.

### DISCUSSION THREAD

**Discussion by khoroshyy** [michaelmarty/UniDec#61] [Q&A]
If I run batch run directories, is it possible to save the deconvoluted spectrum as a txt file? Not only picked peaks.
Thanks.

**Comment by michaelmarty** [comment]
Yes, it should be saved in the _mass.txt file in the unidecfiles folder. See if you can find it there, and let me know if you can't. Thanks!

**Reply by khoroshyy** [reply]
If I run Tools->Batch Run Directories (Waters/Agilent)
and select a few raw directories, for each I get hdf5 file, and in the Unidec_Figures_and_files directory for each raw file I get saved only ...-peaks.txt and ..._extracts.txt files no ..._mass.txt.

**Reply by michaelmarty** [reply]
Ok, that makes sense. You must be using UniChrom. The mass data is stored in the HDF5 file, and there are several ways to get it out. I added a tutorial in the wiki to explain. Take a look here: [https://github.com/michaelmarty/UniDec/wiki/What-Are-All-Those-File-Outputs-and-How-Can-I-Export-My-Data%3F#metaunidec-is-different](https://github.com/michaelmarty/UniDec/wiki/What-Are-All-Those-File-Outputs-and-How-Can-I-Export-My-Data%3F#metaunidec-is-different). Let me know if you have any questions on how this works. I'm working on improving the wiki, so I'm trying to add as much detail there as I can. Thanks!

**Reply by khoroshyy** [reply]
Hi. Thanks for the reply. 
I understand, I can write a script that will extract the data from HDF5, but it will be used by people from the MS facility, and for them, fewer steps would be better. If you can show me the place where the sawing happens, I think I can add a few lines there to be able to save the deconvoluted spectrum too.
 Best.
Petro.

**Reply by michaelmarty** [reply]
Hi Petro,
Sorry for the very delayed response. This got buried as the semester got busy. The place to look the the mudstruct file: https://github.com/michaelmarty/UniDec/blob/master/metaunidec/mudstruct.py. You can also use Ctrl+u on a plot to open a save dialog (https://github.com/michaelmarty/UniDec/wiki/Hidden-Features#export-data-from-plot-to-text-file), but it sounds like you want it to be more automated. The easiest thing would just be drop it in the write_hdf5 function for the spectrum object. Thanks, and again sorry for the delay!
MTM

**Reply by khoroshyy** [reply]
Thanks for the reply. For now, this project I put on hold. Guys got some new hardware and software, so they are busy with that.
